### PR TITLE
imxrt1050-evk/imxrt1050-evk_download.sh: Add support to erase/flash i…

### PR DESCRIPTION
**Pull request by btheosam**
…ndividual partition/binary.

Validate 'make download command' before acutally performing the command.
Add erase option to erase partition(s)
Support download ALL or individual parition/binary

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [969b4a48d6c544e95df23c6ed4e0061cb09998a6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [969b4a48d6c544e95df23c6ed4e0061cb09998a6] - Code Rule Check OK.

File: build/configs/imxrt1050-evk/imxrt1050-evk_download.sh [hunk 1]
```diff
@@ -114,28 +128,79 @@ function get_executable_name()
 	esac
 }	
 
+##Utility function to get partition index ##
+function get_partition_index()
+{
+        case $1 in
+                kernel | Kernel | KERNEL) echo "0";;
+                app | App | APP) echo "1";;
```
Comment (jeongchanKim) [inline build/configs/imxrt1050-evk/imxrt1050-evk_download.sh]: We will not use `app` region anymore.

Comment (sunghan-chang) [reply build/configs/imxrt1050-evk/imxrt1050-evk_download.sh]: This is necessary with protected build.

Comment (jeongchanKim) [reply build/configs/imxrt1050-evk/imxrt1050-evk_download.sh]: Ok. Then how about using user instead of app?

Comment (sunghan-chang) [reply build/configs/imxrt1050-evk/imxrt1050-evk_download.sh]: As we talked on chat, I agree but it is different kind of change. Let's do that with another PR.

**Pull request merged**